tests/e2e/lib/virt_helpers.go

```diff
@@ -0,0 +1,408 @@
+	}
+	log.Printf("Current operator version is: %s", versionString)
+
+	version, err := version.ParseGeneric(versionString)
+	if err != nil {
+		return "", nil, err
+	}
+
+	return csv, version, nil
+}
+
+// Checks the existence of the operator's target namespace
+func (v *VirtOperator) checkNamespace() bool {
+	// First check that the namespace exists
+	exists, _ := DoesNamespaceExist(v.Clientset, v.Namespace)
+	return exists
+}
+
+// Checks for the existence of the virtualization operator group
+func (v *VirtOperator) checkOperatorGroup() bool {
+	group := operatorsv1.OperatorGroup{}
+	err := v.Client.Get(context.TODO(), client.ObjectKey{Namespace: v.Namespace, Name: "kubevirt-hyperconverged-group"}, &group)
+	if err != nil {
+		return false
+	}
+	return true
+}
+
+// Checks if there is a virtualization subscription
+func (v *VirtOperator) checkSubscription() bool {
+	subscription := operatorsv1alpha1.Subscription{}
+	err := v.Client.Get(context.TODO(), client.ObjectKey{Namespace: v.Namespace, Name: "hco-operatorhub"}, &subscription)
+	if err != nil {
+		return false
+	}
+	return true
+}
+
+// Checks if the ClusterServiceVersion status has changed to ready
+func (v *VirtOperator) checkCsv() bool {
+	subscription, err := v.GetOperatorSubscription()
+	if err != nil {
+		if err.Error() == "no subscription found" {
+			return false
+		}
+	}
+
+	return subscription.CsvIsReady(v.Client)
+}
+
+// CheckHco looks for a HyperConvergedOperator and returns whether or not its
+// health status field is "healthy". Uses dynamic client to avoid uprooting lots
+// of package dependencies, which should probably be fixed later.
+func (v *VirtOperator) checkHco() bool {
+	resourceId := schema.GroupVersionResource{
+		Group:    "hco.kubevirt.io",
+		Resource: "hyperconvergeds",
+		Version:  "v1beta1",
+	}
+
+	unstructuredHco, err := v.Dynamic.Resource(resourceId).Namespace(v.Namespace).Get(context.Background(), "kubevirt-hyperconverged", v1.GetOptions{})
+	if err != nil {
+		log.Printf("Error getting HCO: %v", err)
+		return false
+	}
+
+	health, ok, err := unstructured.NestedString(unstructuredHco.UnstructuredContent(), "status", "systemHealthStatus")
+	if err != nil {
+		log.Printf("Error getting HCO health: %v", err)
+		return false
+	}
+	if !ok {
+		log.Printf("HCO health field not populated yet")
+		return false
+	}
+	log.Printf("HCO health status is: %s", health)
+
+	return health == "healthy"
+}
+
+// Creates the target virtualization namespace, likely openshift-cnv or kubevirt-hyperconverged
+func (v *VirtOperator) installNamespace() error {
+	err := v.Client.Create(context.Background(), &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: v.Namespace}})
+	if err != nil {
+		log.Printf("Failed to create namespace %s: %v", v.Namespace, err)
+		return err
+	}
+	return nil
+}
+
+// Creates the virtualization operator group
+func (v *VirtOperator) installOperatorGroup() error {
```

Comment:
Same comment. I like simplicity of installOperatorGroup function, but possibly it should be just calling onother common method with name/namespace. Those functions are accessible within `&operatorsv1.*` but to simplify our code would be great to have such call in one place and the `installOperatorGroup` just a minimum required metadata to install such group. We may us then such call for other operatorgroups.

Reply:
Yes, having this defer to a common CreateOperatorGroup would be nice.